plugins/orchestrator-backend/src/service/SonataFlowService.ts

```diff
@@ -104,15 +106,21 @@ export class SonataFlowService {
         ? `${args.serviceUrl}/${args.definitionId}?businessKey=${args.businessKey}`
         : `${args.serviceUrl}/${args.definitionId}`;
 
-      const result = await fetch(urlToFetch, {
+      const response = await fetch(urlToFetch, {
         method: 'POST',
         body: JSON.stringify(args.inputData),
         headers: { 'content-type': 'application/json' },
       });
 
-      const json = await result.json();
-      this.logger.debug(`Execute workflow result: ${JSON.stringify(json)}`);
-      return json;
+      if (response.ok) {
+        const json = await response.json();
+        this.logger.debug(`Execute workflow result: ${JSON.stringify(json)}`);
+        return json;
+      }
+      const res = await response.json();
+      throw new Error(
+        `${this.createPrefixFetchErrorMessage(urlToFetch, 'executeWorkflow', 'POST')} - Details: ${res?.details || 'No details provided'}, Stack: ${res?.stack || 'No stack trace available'}`,
```

Comment:
you don't need to do `Stack: ${res?.stack || 'No stack trace available'}`,` leaving the empty string or object is fine too, if there is nothing in it, the user will understand there is no stack trace
Or simply do not add the `Stack` in the error message if there is none

Reply:
I added for more clarity, I think `No stack trace/details available` is clearer than empty.
But I can remove it

Reply:
Why saying something is not there? If it's not there, it's not there, no need to add "noise" (and use some bytes)
From an UI perspective, I get it why you would print something like that, but this error will be catch and used internally before being displayed to the end user I think :)

Reply:
Can you please check if it looks ok now?